wagtailmedia/static/wagtailmedia/js/WagtailMediaBlock.js

```diff
@@ -0,0 +1,206 @@
+
+/**
+ *	Places media thumbnail HTML in the Rich Text Editor
+ */
+class WagtailMediaBlock extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showTooltipAt: null,
+        };
+
+        this.setState      = this.setState.bind(this);
+        this.openTooltip   = this.openTooltip.bind(this);
+        this.closeTooltip  = this.closeTooltip.bind(this);
+        this.renderTooltip = this.renderTooltip.bind(this);
+    }
+
+    componentDidMount() {
+        document.addEventListener('mouseup', this.closeTooltip);
+        document.addEventListener('keyup', this.closeTooltip);
+        window.addEventListener('resize', this.closeTooltip);
+    }
+
+    openTooltip(e) {
+        const { blockProps } = this.props;
+        const { entity, onRemoveEntity } = blockProps;
+        const data = entity.getData();
+
+        const trigger = e.target.closest('[data-draftail-trigger]');
+
+        if (!trigger) return; // Click is within the tooltip
+
+        const container = trigger.closest('[data-draftail-editor-wrapper]');
+
+        if (container.children.length > 1) return;  // Tooltip already exists
+
+        const containerRect = container.getBoundingClientRect();
+        const rect = trigger.getBoundingClientRect();
+        const maxWidth = trigger.parentNode.offsetWidth - rect.width;
+        const direction = maxWidth >= TOOLTIP_MAX_WIDTH ? 'left' : 'top-left'; // Determines position of the arrow on the tooltip
+
+        let top  = 0;
+        let left = 0;
+
+        if(direction == 'left'){
+            left = rect.width + 50;
+            top  = rect.top - containerRect.top + (rect.height / 2);
+        }
+        else if (direction == 'top-left'){
+            top = rect.top - containerRect.top + rect.height;
+        }
+
+        this.setState({
+            showTooltipAt: {
+                container: container,
+                top: top,
+                left: left,
+                width: rect.width,
+                height: rect.height,
+                direction: direction,
+            }
+        });
+    }
+
+    closeTooltip(e) {
+        if(e.target.classList){
+            if(e.target.classList.contains("Tooltip__button")){
+                return; // Don't setState if the "Delete" button was clicked
+            }
+        }
+        this.setState({ showTooltipAt: null });
+    }
+
+    /**
+     * Returns either a tooltip "portal" element or null
+     */
+    renderTooltip(data) {
+        const { showTooltipAt } = this.state;
+        const { blockProps } = this.props;
+        const { entity, onRemoveEntity } = blockProps;
+
+        // No tooltip coords exist, don't show one
+        if(!showTooltipAt) return null;
+
+        let options = []
+        if(data.autoplay) options.push("Autoplay");
+        if(data.mute) options.push("Mute");
+        if(data.loop) options.push("Loop");
+        const options_str = options.length ? options.join(", ") : "";
```

Comment:
Not to worry about right now but these strings should be translate-able. We can implement this similarly to Wagtail.